examples/Raytracer_Benchmark/Navigator.h

```diff
@@ -0,0 +1,245 @@
+// SPDX-FileCopyrightText: 2020 CERN
+// SPDX-License-Identifier: Apache-2.0
+
+/**
+ * @file Navigator.h
+ * @brief Navigation methods for geometry.
+ */
+
+#ifndef RT_NAVIGATOR_H_
+#define RT_NAVIGATOR_H_
+
+#include <CopCore/Global.h>
+
+#include <VecGeom/base/Global.h>
+#include <VecGeom/base/Vector3D.h>
+#include <VecGeom/management/BVHManager.h>
+#include <VecGeom/navigation/NavStateIndex.h>
+
+#ifdef VECGEOM_ENABLE_CUDA
+#include <VecGeom/backend/cuda/Interface.h>
+#endif
+
+inline namespace COPCORE_IMPL {
+
+class Navigator {
+
+public:
+  using VPlacedVolumePtr_t = vecgeom::VPlacedVolume const *;
+
+  __host__ __device__
+  static VPlacedVolumePtr_t LocatePointIn(vecgeom::VPlacedVolume const *vol,
+                                          vecgeom::Vector3D<vecgeom::Precision> const &point,
+                                          vecgeom::NavStateIndex &path, bool top)
+  {
+    if (top) {
+      assert(vol != nullptr);
+      if (!vol->UnplacedContains(point)) return nullptr;
+    }
```

Comment:
Why does it work on trackML.gdml and not cms2018.gdml? I guess the answer is that the origin point of the camera is such that for trackML the point is outside the world bounding box, while the CMS geometry is bigger, so the point lies inside the geometry and the first call to the navigator moves to the rest of the body, which then tries to access uninitialized stuff...

Reply:
Probably. Have you tried to call the program with -screenx putting a larger negative value (now the default is -5000)? This should not crash of course. Can vecgeom::cxx be made explicit in the code at that place (since the call is supposed to be on the host geometry) or it just does not compile?

Reply:
Yes, when I call with a point far away, it gets past the point where it broke before. For CMS, I used `-screeny -20000`.